### PR TITLE
Start work on Migration REST Resource.

### PR DESCRIPTION
**Pull request by LukasMansour**
Motivation:
We need REST APIs for the KAI Projects.

Modification:
Adds and registeres a MigrationResource with which migrations can be started on pools.

Result:
The dcache-frontend now contains a MigrationResource which allows pools to be migrated, the Rest API forwards the 'migration copy' command to the designated pool and returns the migration job id in the response header.

Signed-off-by: Lukas Mansour [lukas.mansour@desy.de](mailto:lukas.mansour@desy.de)

### CODE REVIEW

**Comment by kofemann** [comment]
ok to test

**Comment by LukasMansour** [comment]
I believe this is ready for a first review.

File: modules/dcache-frontend/src/main/java/org/dcache/restful/resources/migration/MigrationResources.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+            return Response.status(Response.Status.CREATED)
+                  .header("migration-job-id", migrationJobId).type(MediaType.APPLICATION_JSON)
+                  .build();
+        } catch (InterruptedException | CacheException | NoRouteToCellException e) {
+            throw new InternalServerErrorException(e);
+        }
+
+    }
+
+    private static void checkAppendString(String cmdParam, String jsonKey, StringBuilder sb,
```
Comment (kofemann) [inline modules/dcache-frontend/src/main/java/org/dcache/restful/resources/migration/MigrationResources.java:239]: let call in `conditionalAppend` as check typically throws an exception if check failes

**Review by kofemann — APPROVED** [review]
👍🏼 good to go in

File: modules/dcache-frontend/src/main/java/org/dcache/restful/resources/migration/MigrationResources.java (hunk 1 above)
Comment (LukasMansour) [reply modules/dcache-frontend/src/main/java/org/dcache/restful/resources/migration/MigrationResources.java:239]: Okay sounds good, will rename

**Comment by kofemann** [comment]
🎉

**Pull request merged**